goto_handler.go

```diff
@@ -0,0 +1,116 @@
+package migrate
+
+import (
+	"fmt"
+	"log"
+	"os"
+	"path/filepath"
+	"strconv"
+	"strings"
+)
+
+func (m *Migrate) HandleDirtyState(path string) {
+	// Perform actions when dirty is true
+		lastSuccessfulMigrationPath := filepath.Join(path, "lastSuccessfulmigration")
+		lastVersionBytes, err := os.ReadFile(lastSuccessfulMigrationPath)
+		if err != nil {
+			log.Fatal(fmt.Errorf("failed to read last successful migration file: %s", err))
```

Comment:
You should be able to use https://pkg.go.dev/log#Fatalf and get rid of `fmt.Errorf()`.